ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -531,7 +531,15 @@ func (r *RayServiceReconciler) shouldPrepareNewRayCluster(ctx context.Context, r
 			return RolloutNew
 		}
 
-		// Case 1: If everything is identical except for the Replicas and WorkersToDelete of
+		// Case 1: If the KubeRay version has changed and the hashes are not identical, update the RayCluster to get
```

Comment:
This comment implies the hash needs to be checked, but we return  `Update` before checking the hash, is that expected?

Reply:
Oh I forgot to change the comment here (this check used to be after we checked that the hashes are different). Whether or not the hash is different, I think we want to just return `Update` on this iteration to get the correct KubeRay version and hash (if it has actually changed).